Source/WebCore/loader/ContentFilter.cpp

```diff
@@ -123,6 +123,64 @@ bool ContentFilter::continueAfterWillSendRequest(ResourceRequest& request, const
     return !request.isNull();
 }
 
+void ContentFilter::continueAfterWillSendRequest(ResourceRequest&& request, const ResourceResponse& redirectResponse, CompletionHandler<void(ResourceRequest)>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+
+    // Return early if no platform content filters are enabled.
+    if (!anyContentFilterIsEnabled()) {
+        continueAfterWillSendRequest(request, redirectResponse);
+        completionHandler(WTFMove(request));
+        return;
+    }
+
+    LOG(ContentFiltering, "ContentFilter received request for <%{sensitive}s> with redirect response from <%{sensitive}s>.\n", request.url().string().ascii().data(), redirectResponse.url().string().ascii().data());
+    ASSERT(m_state == State::Stopped || m_state == State::Filtering);
+
+    m_platformCompletionHandlersCalled = 0;
+    m_platformFiltersAllowed = 0;
+    m_willSendRequestCompletionHandler = WTFMove(completionHandler);
+
+    for (Ref platformContentFilter : m_contentFilters) {
+
+        URL originalURL = request.url();
+        CompletionHandler<void(ResourceRequest, PlatformContentFilter::State)> completion = [platformContentFilter, protectedThis = Ref { *this }, originalURL = WTFMove(originalURL)] (ResourceRequest request, PlatformContentFilter::State state) mutable {
+            ASSERT(RunLoop::isMain());
+            ++protectedThis->m_platformCompletionHandlersCalled;
```

Comment:
Is there a way to leverage `WTF::CallbackAggregator` instead of dealing with counts? Seems it would be more robust.